### PR TITLE
[GDR-2530] Switch to get_supported_experiments

### PR DESCRIPTION
**Pull request by j-smola**
# Description
## What changed?
Related JIRA issue: [GDR-2530](https://jira.gene.com/jira/browse/GDR-2530)

## Why was it changed?

# Checklist for sustainable code base
- [ ] I added tests for any code changed/added
- [ ] I added documentation for any code changed/added
- [ ] I made sure naming of any new functions is self-explanatory and consistent

# Logistic checklist
- [x] Package version bumped
- [x] Changelog updated

# Screenshots (optional)

### CODE REVIEW

File: R/runDrugResponseProcessingPipeline.R [hunk 1]
```diff
@@ -234,8 +234,7 @@ runDrugResponseProcessingPipeline <- function(x,
   checkmate::assert_choice(start_from, get_pipeline_steps())
   checkmate::assert_character(selected_experiments, null.ok = TRUE)
   checkmate::assert_subset(selected_experiments,
-                           names(gDRutils::get_experiment_groups()),
-  )
+                           gDRutils::get_supported_experiments())
```
Comment (j-smola) [inline R/runDrugResponseProcessingPipeline.R:237]: @bczech Please, confirm that this change won't cause any issue.
previous
```
> names(gDRutils::get_experiment_groups())
[1] "single-agent" "combination" 
```
current
```
> gDRutils::get_supported_experiments()
[1] "single-agent" "combination"  "co-dilution" 
```

File: R/utils.R [hunk 2]
```diff
@@ -228,22 +217,27 @@ get_data_type_to_data_model_mapping <- function() {
...
 #' @export
 data_model.character <- function(x) {
-  # TODO: switch to gDRutils::get_experiment_groups()
-  # once we clean-up single-agent/combination assignemnts
-  exp_v <- get_data_type_to_data_model_mapping()
-  
-  checkmate::assert_subset(x, names(exp_v))
-  
-  exp_v[[x]]
+  checkmate::assert_subset(x, gDRutils::get_supported_experiments())
```
Comment (j-smola) [inline R/utils.R:220]: @gladkia Please, check and confirm that it is what you meant (get rid of `get_data_type_to_data_model_mapping()` with refactor `data_model.character()` and `validate_data_models_availability()`

File: R/runDrugResponseProcessingPipeline.R (hunk 1 above)
Comment (bczech) [reply R/runDrugResponseProcessingPipeline.R:237]: it's ok

**Review by bczech — APPROVED** [review]

File: R/utils.R [hunk 3]
```diff
@@ -228,22 +217,27 @@ get_data_type_to_data_model_mapping <- function() {
...
+  checkmate::assert_character(d_types)
+  checkmate::assert_character(s_d_models, null.ok = TRUE)
   
-  dm_v <- get_data_type_to_data_model_mapping()
-  
+  dm_v <- gDRutils::get_experiment_groups()
+
   req_d_models <-
-    unique(as.character(dm_v[names(dm_v) %in% d_types]))
+    unique(names(dm_v)[vapply(d_types, function(i) grep(i, dm_v), integer(1))])
```
Comment (gladkia) [inline R/utils.R:240]: Is the `grep` needed here?

File: R/utils.R (hunk 2 above)
Comment (gladkia) [reply R/utils.R:220]: @bczech can we assume that the data model is now simply the experiment name? I mean for `co-dilution` the data model will be the `co-dilution`. If this is the case the logic can be even more simplified.

File: R/runDrugResponseProcessingPipeline.R (hunk 1 above)
Comment (j-smola) [reply R/runDrugResponseProcessingPipeline.R:237]: ✔️

File: R/utils.R (hunk 3 above)
Comment (j-smola) [reply R/utils.R:240]: @gladkia  Could you proposed other solution?

Comment (j-smola) [reply R/utils.R:240]:
```
exp <- gDRutils::get_supported_experiments()
exp
dm_v <- gDRutils::get_experiment_groups()
dm_v

d_types <- exp[1]
d_types
req_d_models <-
  unique(names(dm_v)[vapply(d_types, function(i) grep(i, dm_v), integer(1))])
req_d_models

d_types <- c(exp[2], exp[3])
req_d_models <-
  unique(names(dm_v)[vapply(d_types, function(i) grep(i, dm_v), integer(1))])
req_d_models

d_types <- c(exp[1], exp[3])
req_d_models <-
  unique(names(dm_v)[vapply(d_types, function(i) grep(i, dm_v), integer(1))])
req_d_models
```

Comment (bczech) [reply R/utils.R:240]: @j-smola you can get rid of grep by using e.g. `unique(names(dm_v)[vapply(dm_v, function(x) any(d_types %in% x), logical(1))])`

Comment (bczech) [reply R/utils.R:240]: or shorter without `vapply` `unique(names(Filter(function(x) any(d_types %in% x), dm_v)))`

Comment (j-smola) [reply R/utils.R]: Changed into `any` :)

File: R/utils.R (hunk 2 above)
Comment (j-smola) [reply R/utils.R:220]: Continuation in [GDR-2542](https://jira.gene.com/jira/browse/GDR-2542)

**Review by gladkia — APPROVED** [review]

**Pull request merged**